src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateObject = {};
+  const separator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const splitedDate = date.split(separator);
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = splitedDate[i];
+  }
+
+  if (dateObject.hasOwnProperty('YYYY')) {
+    dateObject.YY = dateObject.YYYY.slice(2);
+  }
```

Comment:
The conversion from 'YYYY' to 'YY' is done unconditionally. You should check if 'YY' is needed in the 'toFormat' before slicing 'YYYY'. Otherwise, you might be doing unnecessary work.